core/src/mindustry/editor/WaveInfoDialog.java

```diff
@@ -41,6 +46,40 @@ public WaveInfoDialog(){
         addCloseListener();
 
         onResize(this::setup);
+        buttons.button(Icon.filter, () -> {
+            BaseDialog dialog = new BaseDialog("@waves.sort");
+            dialog.setFillParent(false);
+            dialog.cont.defaults().size(210f, 64f);
+            dialog.cont.button("@waves.sort.begin", () -> {
```

Comment:
This is pretty much the exact same code duplicated 4 times; loop over every sort mode and add a button for it instead.

Reply:
saved a lot of codes, thank :)